packages/core/src/Stores/client-store.js

```diff
@@ -1963,8 +1963,15 @@ export default class ClientStore extends BaseStore {
         }
 
         const is_client_logging_in = login_new_user ? login_new_user.token1 : obj_params.token1;
+
         if (is_client_logging_in) {
-            window.history.replaceState({}, document.title, sessionStorage.getItem('redirect_url'));
+            const DERIV_PRE_APPSTORE_KEY = 'is_pre_appstore';
+            const is_pre_appstore = window.localStorage.getItem(DERIV_PRE_APPSTORE_KEY);
+            if (is_pre_appstore === 'true') {
```

Comment:
check the `redirect_url` to be trader page( `/`) then push to trading-hub page, else ...